apps/language_server/test/providers/execute_command/manipulate_pipes/ast_test.exs

```diff
@@ -0,0 +1,159 @@
+defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipes.ASTTest do
+  use ExUnit.Case
+
+  alias ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipes.AST
+
+  describe "to_pipe/1" do
+    test "single-line selection with two args in named function" do
+      assert "A.B.C.a |> X.Y.Z.function_name(b)" == AST.to_pipe("X.Y.Z.function_name(A.B.C.a, b)")
+    end
+
+    test "single-line selection with single arg in named function" do
+      assert "A.B.C.a |> X.Y.Z.function_name()" == AST.to_pipe("X.Y.Z.function_name(A.B.C.a)")
+    end
+
+    test "single-line selection with two args in anonymous function" do
+      assert "A.B.C.a |> X.Y.Z.function_name.(b)" ==
+               AST.to_pipe("X.Y.Z.function_name.(A.B.C.a, b)")
+    end
+
+    test "single-line selection with single arg in anonymous function" do
+      assert "A.B.C.a |> function_name.()" == AST.to_pipe("function_name.(A.B.C.a)")
+    end
+
+    test "multi-line selection with two args in named function" do
+      assert "X.Y.Z.a |> X.Y.Z.function_name(b, c)" ==
+               AST.to_pipe("""
+                 X.Y.Z.function_name(
+                 X.Y.Z.a,
+                 b,
+                 c
+               )
+               """)
```

Comment:
```suggestion          
      actual =
        AST.to_pipe("""
        X.Y.Z.function_name(
          X.Y.Z.a,
          b,
          c
          )
        """)

      expected =
        if Version.match?(System.version(), "< 1.10.0") do
          "X.Y.Z.a() |> X.Y.Z.function_name(b, c)"
        else
          "X.Y.Z.a |> X.Y.Z.function_name(b, c)"
        end

      assert actual == expected
```
We should change the Elixir version-dependant tests to use different assertions for tests that vary based on the Elixir version. But in general we should strive to not have tests that are specific to an Elixir version. I need to look at the rest of the code but could we generate `X.Y.Z.a()` version for all elixir versions? I feel that might be slightly more idiomatic (although since the formatter will accept either `X.Y.Z.a()` or `X.Y.Z.a` it is a little ambiguous.

Reply:
I think that changing the tests cases for having `X.Y.Z.a()` as the argument would fix that.
Either that, or forcing parens or no parens throught the AST module would remove the need for making the tests version-dependent